src/H5Odtype.c

```diff
@@ -24,6 +24,15 @@
 #include "H5Tpkg.h"      /* Datatypes            */
 #include "H5VMprivate.h" /* Vectors and arrays   */
 
+/* Variant boundary-checking macro, used here since H5Tdecode() doesn't take a
+ * size parameter so we need to ignore the bounds checks.
+ *
+ * This is a separate macro since we don't want to inflict that behavior on
+ * the rest of the library.
+ */
+#define H5_DTYPE_IS_BUFFER_OVERFLOW(skip, ptr, size, buffer_end)                                             \
+    (skip ? FALSE : ((ptr) + (size)-1) > (buffer_end))
+
```

Comment:
Ugly hack, but necessary to work around H5Tencode()'s deficiencies. I'll probably version that call to take a size parameter in 2.0.0.